[24.2] Reduce default framework tool test timeout to 60 seconds

mvdbeek opened the pull request:
This seems like a reasonable value for tests run via GalaxyTestDriver (integration, api, but not galaxy-tool-util / planemo).
At a minimum we don't burn through our minute quota when tests time out indefinitely, like the rucio tests do currently.

Also fixes neo4j converter test definition and mocks out doi.org requests.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

mvdbeek committed:
Reduce default framework tool test timeout to 60 seconds

mvdbeek committed:
Mock unit test doi.org requests

We're testing the cache, not the requests here.

mvdbeek committed:
Make bgzip a CompressedArchive subclass

mvdbeek committed:
Fix neostore.zip converter tests

nsoranzo commented on test/unit/app/managers/test_CitationsManager_db.py:
```diff
@@ -27,16 +25,17 @@ def __init__(self, config, db_url):
+        for doi, author in dois:
+            responses.add(method="GET", url=f"https://doi.org/{doi}", body=author)
+        assert "Jörg" in doi_cache.get_bibtex("10.1093/bioinformatics/bts252")
+        assert "Özkurt" in doi_cache.get_bibtex("10.1101/2021.12.24.474111")
```
```suggestion
            assert author in doi_cache.get_bibtex(doi)
```

mvdbeek committed:
Improve test assertion

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

mvdbeek committed:
Extend tool test timeout for pulsar resubmission

mvdbeek committed:
Drop size assertions on tabix index

That's none of our business. We know index is valid if the chromosome
names are printed to stdout.

nsoranzo committed:
Store maxseconds on GalaxyTestDriver

mvdbeek committed:
Lower pulsar transport_timeout param instead of extending wait
